### PR TITLE
fix(*): Optimize selected account validation

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
@ajordens thanks for your fix in #3608 and #3609! Sorry about that :/ I have a minor update to optimize the code and added some tests to prevent future regressions

I verified this fix on the edit application dialog and on some Azure dialogs for creating resources (which prompted my original change) and it looks good.

@anotherchrisberry PTAL as well

### CODE REVIEW

**Comment by ajordens** [comment]
Thanks for the followup @EricJizbaMSFT. 

I had originally gone down the `_.intersection` path and saw some more tasty lodash to drink :)

I'll defer to @anotherchrisberry on the specifics but it seems reasonable to me.

File: app/scripts/modules/core/account/accountSelectField.directive.js [hunk 1]
```diff
@@ -56,10 +56,10 @@ module.exports = angular
         }
 
         if (this.component) {
-          var mergedAccounts = (this.mergedAccounts || []).sort();
-          var component = _.flatten([this.component[this.field]]).sort();
+          var mergedAccounts = this.mergedAccounts || [];
+          var component = _.flatten([this.component[this.field]]);
 
-          if (!_.isEqual(_.intersection(mergedAccounts, component), component)) {
+          if (_.intersection(mergedAccounts, component).length != component.length) {
```
Comment (ajordens) [inline app/scripts/modules/core/account/accountSelectField.directive.js]: should we `!==` because javascript?

**Comment by anotherchrisberry** [comment]
LGTM - just needs a rebase

**Comment by EricJizbaMSFT** [comment]
@anotherchrisberry rebase done

**Comment by anotherchrisberry** [comment]
one more rebase and we'll get this merged

**Comment by EricJizbaMSFT** [comment]
done

**Pull request merged**